<launch>
    <arg name="image_topic" default="/diy_img/rgb/image_raw"/>
    <arg name="depth_topic" default="/diy_img/depth/image_raw"/>
    <arg name="odom_topic" default="/diy_odom"/>
    <arg name="use_rviz" default="true"/>

    <!-- 启动 d2p_node -->
    <node name="d2p_node" pkg="d2p" type="d2p_node" output="screen">
        <remap from="/iris_0/realsense/depth_camera/depth/image_raw" to="$(arg depth_topic)"/>
        <remap from="/iris_0/mavros/vision_odom/odom" to="$(arg odom_topic)"/>
    </node>
    <node name="detect_node" pkg="d2p" type="yolo_detector_node.py" output="screen">
        <param name="image_topic" value="$(arg image_topic)"/>
    </node>
    <group if="$(arg use_rviz)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find d2p)/rviz/detector.rviz"/>
    </group>
</launch>
